client.go

```diff
@@ -133,7 +135,7 @@ func NewClientWithConfig(cfg *Config) (*Client, error) {
 	if len(cfg.Network.URLs) == 0 {
 		return nil, fmt.Errorf("at least one url should be present in config in 'secret_urls = []'")
 	}
-	tr, err := NewTracer(cfg.Network.URLs[0], cs, &abiFinder, cfg, contractAddressToNameMap, addrs)
+	tr, err := NewTracer(cfg.Network.URLs[0], cfg.RPCHeaders, cs, &abiFinder, cfg, contractAddressToNameMap, addrs)
```

Comment:
wdyt about just passing cfg here? we could seriously reduce the number of params

Reply:
done